content-scripts/index.ts

```diff
@@ -0,0 +1,160 @@
+import {
+    injectOverrideFunctions,
+    reverseDateDirection,
+    currentAccountPage,
+    addSortTableActions,
+} from "./modules/initialize";
+import { injectAllChanges, userPreferences } from "./modules/options";
+import constructNewData from "./modules/utilities/constructNewData";
+import { getStorage, setStorage } from "./modules/utilities/storage";
+import { rememberLogin } from "./modules/login";
+import { replaceIcons } from "./modules/icons";
+import { teacherPage } from "./pages/teacher_page";
+import { classPage } from "./pages/class_page";
+import { improveSchedule } from "./modules/schedule";
+import { changeCourseCards, changeProfileRow } from "./pages/profile_page";
+import { courseUnitPage } from "./pages/course_unit_page";
+import { fixPagination } from "./modules/pagination";
+import { changeLayout } from "./modules/layout";
+import { addStarIconToCard } from "./modules/favorite-course";
+import { createComponentsPage } from "./pages/components_page";
+
+const GENERIC_SIGARRA_URL: string = "^https://sigarra.up.pt/feup/(pt|en)/";
+const BASE_SIGARRA_URL: string = GENERIC_SIGARRA_URL + "(?!.*/components)";
+const COMPONENTS_URL: string = GENERIC_SIGARRA_URL + "(?=.*/components)";
```

Comment:
because of `?=.*/` the `https://sigarra.up.pt/feup/pt/components` does not work

Reply:
i wasn't able to get `https://sigarra.up.pt/feup/pt/components` to work, however since this is mostly a debugging/documentation tool intended to be used by the team, it might be okay. what do you think?

Reply:
humn. Personally, I prefer to access ` https://sigarra.up.pt/feup/pt/components` rather thatn `https://sigarra.up.pt/feup/pt/adasdasd/components` ahah